### PR TITLE
clang-tidy fixes

### PR DESCRIPTION
**Pull request by neheb**
### Short description
modernize and performance clang-tidy fixes.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by neheb** [comment]
That failure looks bogus.

**Comment by omoerbeek** [comment]
Thanks for the work on this!
I do not look in detail yet, but please use the naming conventions we have. do no `d_xxxxx` for local vars. The `d_` prefix is used for members of a class.
Also, I personally prefer smaller, easier to digest diffs compared to ones trying to fix everything everywhere. With a diff as large as this, it becomes harder to spot a small mistake.

**Comment by neheb** [comment]
Hmm? I’ve split up the changes in multiple commits.

**Comment by neheb** [comment]
I should also mention that most of these changes are automated. The bind one was done manually as clang-tidy uses auto&& which is unsupported in C++11.

**Comment by omoerbeek** [comment]
> Hmm? I’ve split up the changes in multiple commits.

Yes, but the unit of review is a PR.  The issue (for me at least) is that it is harder for me to find time and keep track of the reviews status of one big PR than several smaller ones. I'll talk to the other developers to see what they prefer.

**Comment by omoerbeek** [comment]
Hello @neheb, I still prefer smaller PRs but I plan to take this up anyway. Do you have a chance to rebase this and resolve the conflicts? Thanks.

**Comment by neheb** [comment]
Rebased.

**Review by omoerbeek — COMMENTED** [review]
`ext/yahttp/yahttp` is third party, I suggest to PR the changes to it this upstream (https://github.com/cmouse/yahttp)

**Comment by pieterlexis** [comment]
@neheb Can you rebase this again(sorry) so we can merge?

**Review by omoerbeek — APPROVED** [review]
I did push to this branch to fix my problem with this PR: avoid local vars starting with `d_`. Thanks again and sorry for the delay

File: pdns/ws-auth.cc [hunk 1]
```diff
@@ -177,20 +177,20 @@ void AuthWebServer::printvars(ostringstream &ret)
   ret<<"<div class=panel><h2>Variables</h2><table class=\"data\">"<<endl;
 
   vector<string>entries=S.getEntries();
-  for(vector<string>::const_iterator i=entries.begin();i!=entries.end();++i) {
-    ret<<"<tr><td>"<<*i<<"</td><td>"<<S.read(*i)<<"</td><td>"<<S.getDescrip(*i)<<"</td>"<<endl;
+  for(const auto & entrie : entries) {
```
Comment (rgacogne) [inline pdns/ws-auth.cc:180]:
```suggestion
  for(const auto & entry : entries) {
```
?

File: pdns/dnspacket.cc [hunk 2]
```diff
@@ -166,12 +164,10 @@ vector<DNSZoneRecord*> DNSPacket::getAnswerRecords()
 {
   vector<DNSZoneRecord*> arrs;
 
-  for(vector<DNSZoneRecord>::iterator i=d_rrs.begin();
-      i!=d_rrs.end();
-      ++i)
+  for(auto & d_rr : d_rrs)
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnspacket.cc:167]: this comment was not resolved

File: pdns/dnspacket.cc [hunk 3]
```diff
@@ -528,14 +524,12 @@ try
     if(edo.d_extFlags & EDNSOpts::DNSSECOK)
       d_dnssecOk=true;
 
-    for(vector<pair<uint16_t, string> >::const_iterator iter = edo.d_options.begin();
-        iter != edo.d_options.end(); 
-        ++iter) {
-      if(iter->first == EDNSOptionCode::NSID) {
+    for(const auto & d_option : edo.d_options) {
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/dnspacket.cc:527]: unresolved

**Review by rgacogne — APPROVED** [review]
Nice work! One naming nit, I'm fine with merging this PR as it is.

File: pdns/signingpipe.cc [hunk 4]
```diff
@@ -132,12 +132,12 @@ pair<vector<int>, vector<int> > ChunkedSigningPipe::waitForRW(bool rd, bool wr,
 {
   vector<pollfd> pfds;
 
-  for(unsigned int n = 0; n < d_sockets.size(); ++n) {    
-    if(d_eof.count(d_sockets[n]))  
+  for(int & d_socket : d_sockets) {
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/signingpipe.cc:135]: unresolved

File: pdns/responsestats.cc [hunk 5]
```diff
@@ -63,10 +63,8 @@ map<uint16_t, uint64_t> ResponseStats::getQTypeResponseCounts()
 map<uint16_t, uint64_t> ResponseStats::getSizeResponseCounts()
 {
   map<uint16_t, uint64_t> ret;
-  for(sizecounters_t::const_iterator iter = d_sizecounters.begin();
-      iter != d_sizecounters.end();
-      ++iter) {
-    ret[iter->first]=iter->second;
+  for(const auto & d_sizecounter : d_sizecounters) {
```
Comment (Habbie) [inline pdns/responsestats.cc:66]: `d_`

File: pdns/statbag.cc [hunk 6]
```diff
@@ -373,10 +373,10 @@ string StatBag::getRingTitle(const string &name)
 vector<string>StatBag::listRings()
 {
   vector<string> ret;
-  for(auto i=d_rings.begin();i!=d_rings.end();++i)
-    ret.push_back(i->first);
-  for(auto i=d_comboRings.begin();i!=d_comboRings.end();++i)
-    ret.push_back(i->first);
+  for(auto & d_ring : d_rings)
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/statbag.cc:377]: unresolved

File: pdns/statbag.cc [hunk 7]
```diff
@@ -373,10 +373,10 @@ string StatBag::getRingTitle(const string &name)
...
+    ret.push_back(d_ring.first);
+  for(auto & d_comboRing : d_comboRings)
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/statbag.cc:379]: unresolved

File: pdns/stubresolver.cc [hunk 8]
```diff
@@ -156,10 +156,10 @@ int stubDoResolve(const DNSName& qname, uint16_t qtype, vector<DNSZoneRecord>& r
     if(mdp.d_header.rcode == RCode::ServFail)
       continue;
 
-    for(MOADNSParser::answers_t::const_iterator i=mdp.d_answers.begin(); i!=mdp.d_answers.end(); ++i) {
-      if(i->first.d_place == 1 && i->first.d_type==qtype) {
+    for(const auto & d_answer : mdp.d_answers) {
```
[the remark this answers is not in the document]
Comment (Habbie) [reply pdns/stubresolver.cc:162]: unresolved

**Review by Habbie — APPROVED** [review]
Couple of nits and a question, but merging as is would also work for me.

File: modules/gmysqlbackend/smysql.cc [hunk 9]
```diff
@@ -491,9 +491,9 @@ void SMySQL::connect()
   } while (retry >= 0);
 }
 
-SMySQL::SMySQL(const string &database, const string &host, uint16_t port, const string &msocket, const string &user,
-               const string &password, const string &group, bool setIsolation, unsigned int timeout, bool threadCleanup, bool clientSSL):
-  d_database(database), d_host(host), d_msocket(msocket), d_user(user), d_password(password), d_group(group), d_timeout(timeout), d_port(port), d_setIsolation(setIsolation), d_threadCleanup(threadCleanup), d_clientSSL(clientSSL)
+SMySQL::SMySQL(string database, string host, uint16_t port, string msocket, string user,
+               string password, string group, bool setIsolation, unsigned int timeout, bool threadCleanup, bool clientSSL):
+  d_database(std::move(database)), d_host(std::move(host)), d_msocket(std::move(msocket)), d_user(std::move(user)), d_password(std::move(password)), d_group(std::move(group)), d_timeout(timeout), d_port(port), d_setIsolation(setIsolation), d_threadCleanup(threadCleanup), d_clientSSL(clientSSL)
```
[the remark this answers is not in the document]
Comment (rgacogne) [reply modules/gmysqlbackend/smysql.cc:496]: https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html

[the remark this answers is not in the document]
Comment (omoerbeek) [reply modules/gmysqlbackend/smysql.cc:496]: google modernize-pass-by-value

**Comment by omoerbeek** [comment]
Thanks again. I still think smaller PRs are to be preferred, if only to lower the chances of conflicts and make the review and merge process more smooth.

**Pull request merged**